mne/time_frequency/tfr.py

```diff
@@ -1960,38 +1971,46 @@ class EpochsTFR(_BaseTFR):
     verbose : bool, str, int, or None
         If not None, override default verbose level (see :func:`mne.verbose`
         and :ref:`Logging documentation <tut_logging>` for more).
+    events : ndarray, shape (n_events, 3) | None
+        The events as stored in the Epochs class
+    event_id : dict | None
+        Example: dict(auditory=1, visual=3). They keys can be used to access
+        associated events.
+    metadata : instance of pandas.DataFrame | None
+        A :class:`pandas.DataFrame` containing pertinent information for each
+        trial. See :class:`mne.Epochs` for further details
 
     Attributes
     ----------
     info : instance of Info
         Measurement info.
-
     ch_names : list
         The names of the channels.
-
     data : ndarray, shape (n_epochs, n_channels, n_freqs, n_times)
         The data array.
-
     times : ndarray, shape (n_times,)
         The time values in seconds.
-
     freqs : ndarray, shape (n_freqs,)
         The frequencies in Hz.
-
     comment : string
         Comment on dataset. Can be the condition.
-
     method : str | None, defaults to None
         Comment on the method used to compute the data, e.g., morlet wavelet.
-
+    events : ndarray, shape (n_events, 3) | None
+        Array containing sample information as event_id
+    event_id : dict | None
+        Names of conditions correspond to event_ids
+    metadata : DataFrame, shape (n_events, n_cols) | None
+        DataFrame containing pertinent information for each trial
     Notes
     -----
     .. versionadded:: 0.13.0
     """
 
     @verbose
-    def __init__(self, info, data, times, freqs, comment=None,
-                 method=None, verbose=None):  # noqa: D102
+    def __init__(self, info, data, times, freqs, comment=None, method=None,
+                 verbose=None,  events=None, event_id=None, metadata=None):
```

Comment:
verbose is the exception to what I said in my last review. verbose is always the last parameter.